### PR TITLE
ADD Auto test

### PR DESCRIPTION
**Pull request by pushtheworldllc**
Fully documented and described in detail, you can now automatically get the impedances of all channels and inputs.

### CODE REVIEW

**Comment by teonlamont** [comment]
you should squash e8a273c263778594b5069415b8f5acaf2cf391e6 in your history

File: README.md [hunk 1]
```diff
@@ -108,71 +130,116 @@ Measuring impedance is a vital tool in ensuring great data is collected.
...
 
-When you start measuring for electrode impedance's, a new additional property called `impedanceArray` (indexed 1,2,3,4,5,6,7,8), can be found on the sample object from the emitted by 'sample'.
+To run a complete impedance test:
+
+1. Connect to board
+2. Start streaming
+3. Install the 'impedanceObject'
+4. Call `.impedanceTestAllChannels()`
+
+**Note: Takes up to 5 seconds to start measuring impedance's. There is an unknown number of samples taken. Not always 60!**
```
Comment (teonlamont) [inline README.md]: impedance's --> impedances

Comment (pushtheworldllc) [inline README.md]: Ok so isn't it the electrodes impedances??

**Comment by pushtheworldllc** [comment]
And now there are conflicts :(

File: README.md (hunk 1 above)
Comment (teonlamont) [inline README.md]: yeah, just no need for the apostrophe

File: README.md [hunk 2]
```diff
@@ -108,71 +130,116 @@ Measuring impedance is a vital tool in ensuring great data is collected.
...
 
-To apply the test signal to a specific input of a specific channel (i.e. Channel 2 test signal applied to P input and not N input):
 ```js
-ourBoard.impedanceTestStartChannel(2,true,false);
-```
-
-To stop applying the test signal to a specific channel (i.e. Stop applying signal to channel 2):
-```js
-ourBoard.impedanceTestStopChannel(2);
+{
```
Comment (teonlamont) [inline README.md]: i'm a bit confused about use of this impedanceObject

Comment (pushtheworldllc) [inline README.md]: Ok like on al of it?

**Comment by pushtheworldllc** [comment]
I'm closing this request because i have to re work the interface.... After talking with Joel, additional things need to be done:
-------------------
1. ENH functionality to specify which input you want to test, it does not make sense to always test both.
2. ADD outlier identification, in the impedance samples, there is the potential to get an extreme outlier, such 59/60 samples oscillate around 5k Ohm, and then one sample has 2983456 k Ohms which destroys the average impedance, indicating the channel is bad, when really it can be good.
3. ENH The impedance object needs to output and array of objects, not an object of objects

**Comment by teonlamont** [comment]
I will review the reworked tout de suite when you reopen. I am flying back tomorrow morning and will be aching to review some code ;)

**Pull request closed without merging**